test/feature.go

```diff
@@ -71,31 +77,38 @@ func parsePath(path string) ([]*gherkin.Feature, error) {
 		return parseFeatureDir(path)
 	}
 
-	ft, err := parseFeatureFile(path)
+	newIDFunc := (&messages.Incrementing{}).NewId
+	ft, err := parseFeatureFile(path, newIDFunc)
 	if err != nil {
 		return features, err
 	}
 
 	return append(features, ft), nil
 }
 
-func parseFeatureFile(path string) (*gherkin.Feature, error) {
+func parseFeatureFile(path string, newIDFunc func() string) (*feature, error) {
 	reader, err := os.Open(path)
 	if err != nil {
 		return nil, err
 	}
 	defer reader.Close()
 
 	var buf bytes.Buffer
-	ft, err := gherkin.ParseFeature(io.TeeReader(reader, &buf))
+	ft, err := gherkin.ParseGherkinDocument(io.TeeReader(reader, &buf), newIDFunc)
```

Comment:
What is the point of sending the newIDFunc here? Can we create this function variable within this method before calling to gherkin.ParseGherkingDocument and gherkin.Pickles?

As far I can understand, the "newIDFunc" function is a Gherkin thing, so if we isolate all the Gherkin stuff in this method, the better.

Reply:
as said, I just copied and did not much touch this code ...